From 6798037eb1c55f1dd79c04da313bd6bcd4d2aa9e Mon Sep 17 00:00:00 2001
From: uros <uros@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Fri, 22 Apr 2011 18:43:57 +0000
Subject: [PATCH] 	PR target/48723
 	* config/i386/i386.c (ix86_expand_prologue): Update m->fs.sp_offset
 	after the call to ix86_adjust_stack_and_probe.

testsuite/ChangeLog:

	PR target/48723
	* gcc.target/i386/pr48723.c: New test.



git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_6-branch@172867 138bc75d-0d04-0410-961f-82ee72b054a4

index 6e4bb3b..c50cbaf 100644
--- a/gcc/config/i386/i386.c
+++ b/gcc/config/i386/i386.c
@@ -10575,6 +10575,7 @@ ix86_expand_prologue (void)
       if (STACK_CHECK_MOVING_SP)
 	{
 	  ix86_adjust_stack_and_probe (allocate);
+	  m->fs.sp_offset += allocate;
 	  allocate = 0;
 	}
       else
new file mode 100644
index 0000000..ad10209
--- /dev/null
+++ b/gcc/testsuite/gcc.target/i386/pr48723.c
@@ -0,0 +1,13 @@
+/* { dg-do compile } */
+/* { dg-options "-fstack-check -mavx" } */
+
+struct S0
+{
+  int f0, f1, f2, f3;
+} g_106;
+
+struct S0
+func_99 ()
+{
+  return (g_106);
+}
-- 
1.7.0.4

